ZEP000-build-state-on-followers.md

```diff
@@ -0,0 +1,327 @@
+
+### On becoming Leader
+
+On fail-over the new Leader need to drain the map, before he can switch to normal mode. Furthermore, the dispatcher needs to be reset to the right position. This allows to continue with the current state, without reinstalling any dependencies.
+
+### On becoming Follower
+
+If, the node was previously Leader, then the writer strategy will be replaced to no longer really write the entries. Possible race conditions can be handled on the RAFT level, since it detects whether a follower tries to write an entry. The new follower continuous processing, with the next command (it skips events) and fills with the reader the corresponding map.
+
+## Exporter Records
+
+Instead of sending the exported positions periodically over the wire, via SWIM, we could write the current state to the log. This can then be applied on the follower side to rebuild the exporter state, see related [issue](https://github.com/camunda-cloud/zeebe/issues/7088).
+
+# Prior art
+[prior-art]: #prior-art
+     
+[comment]: <> (      Discuss prior art, both the good and the bad, in relation to this proposal. A few examples of what this can include are:)
+[comment]: <> (      - For language, library, tools, and UI proposals: Does this feature exist in other tools/products and what experience have their community had?)
+[comment]: <> (      - For community proposals: Is this done by some other community and what were their experiences with it?)
+[comment]: <> (      - For other teams: What lessons can we learn from what other communities have done here?)
+[comment]: <> (      - Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.)
+[comment]: <> (      This section is intended to encourage you as an author to think about the lessons from other languages, provide readers of your ZEP with a fuller picture. If there is no prior art, that is fine - your ideas are interesting to us whether they are brand new or if it is an adaptation from other languages.)
+[comment]: <> (      Note that while precedent set by other products is some motivation, it does not on its own motivate a ZEP.)
+
+There are several raft implementations out there and all of them doing it in the same way, the state machine is running on all nodes and build the state and take snapshot and compact the log independently.
+
+Other implementations:
+
+  * Hashicorp raft https://github.com/hashicorp/raft
+  * etcd raft https://github.com/etcd-io/etcd/tree/master/raft
+  * dragonboat https://github.com/lni/dragonboat
+
+We also saw [above](#distinction-to-normal-raft) the Raft paper, which mentions that this is the way how we should implement it.
+
+# Out of scope
+[out-of-scope]: #out-of-scope
+
+[comment]: <> (Call out anything which is explicitly not part of this ZEP.)
+
+Building and supporting large states was not covered by this ZEP.
+
+# Unresolved questions
```

Comment:
What will the performance characteristics be?

Reply:
Should I add this as question?